atlasdb-backup/src/main/java/com/palantir/atlasdb/backup/AtlasBackupService.java

```diff
@@ -109,16 +141,44 @@ public static AtlasBackupService createForTests(
     private void storeBackupToken(InProgressBackupToken backupToken) {
         inProgressBackups.put(backupToken.getNamespace(), backupToken);
         backupPersister.storeImmutableTimestamp(backupToken);
+        lockRefresher.registerLocks(ImmutableSet.of(backupToken));
     }
 
     public Set<Namespace> completeBackup(Set<Namespace> namespaces) {
         Set<InProgressBackupToken> tokens = namespaces.stream()
                 .map(inProgressBackups::remove)
                 .filter(Objects::nonNull)
                 .collect(Collectors.toSet());
+        lockRefresher.unregisterLocks(tokens);
+
+        Set<Namespace> namespacesWithInProgressBackups =
+                tokens.stream().map(InProgressBackupToken::getNamespace).collect(Collectors.toSet());
+        if (tokens.size() < namespaces.size()) {
+            Set<Namespace> namespacesWithNoInProgressBackup =
+                    Sets.difference(namespaces, namespacesWithInProgressBackups);
+            log.error(
+                    "In progress backups were not found for some namespaces. We will not complete backup for these.",
+                    SafeArg.of("numNamespacesWithBackup", tokens.size()),
+                    SafeArg.of("numNamespacesWithoutBackup", namespacesWithNoInProgressBackup.size()),
+                    SafeArg.of("namespacesWithoutBackup", namespacesWithNoInProgressBackup),
+                    SafeArg.of("namespaces", namespaces));
+        }
+
         CompleteBackupRequest request = CompleteBackupRequest.of(tokens);
         CompleteBackupResponse response = atlasBackupClient.completeBackup(authHeader, request);
 
+        if (response.getSuccessfulBackups().size() < namespacesWithInProgressBackups.size()) {
```

Comment:
i don't understand this check - we can pair over this

Reply:
happy to.
Briefly, we requested that the client complete backup for some number of namespaces; here, we check if the client successfully completed backup for all of them.